### PR TITLE
[BEAM-2795] Key FlinkRunner streaming translation off URN

### PR DESCRIPTION
**Pull request by kennknowles**
Follow this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/projects/BEAM/issues/) filed for the change (usually before you start working on it).  Trivial changes like typos do not require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [x] Format the pull request title like `[BEAM-XXX] Fixes bug in ApproximateQuantiles`, where you replace `BEAM-XXX` with the appropriate JIRA issue.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [x] Run `mvn clean verify` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [x] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

---

R: @aljoscha this is just the minimal first changes to the streaming translator map. Next will be to extract bits in a raw manner and leave deserialization of DoFns until later.

### CODE REVIEW

**Comment by aljoscha** [comment]
Run Flink ValidatesRunner

**Comment by aljoscha** [comment]
Seems to be a dependency problem.

**Comment by kennknowles** [comment]
OK fixed the deps and checked that locally.

**Comment by kennknowles** [comment]
run Flink ValidatesRunner

**Comment by kennknowles** [comment]
Before kicking this out, I just ran `ParDoTest`, `CombineTest`, and `FlattenTest` in IntelliJ. Looks like there might be shading or dep things going on that doesn't capture.

**Comment by kennknowles** [comment]
Ha, I forgot to run them as streaming tests. The failures were a very silly programming error.

**Comment by kennknowles** [comment]
run flink validatesrunner

**Comment by kennknowles** [comment]
run flink validatesrunner

**Comment by kennknowles** [comment]
OK, the ValidatesRunner tests pass. Please take a look. This is somewhat of an exploratory change; it doesn't actually break dependencies on the Java SDK details. Separating the existing "intrinsic" mode of executing Java SDK pipelines on the FlinkRunner might be helpful in the future to clarify the dependencies.

File: runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTransformTranslators.java [hunk 1]
```diff
@@ -1046,4 +1080,134 @@ public void flatMap(
...
+        SdkComponents components) {
+      throw new UnsupportedOperationException(
+          String.format(
+              "%s should never be translated",
+              SplittableParDoViaKeyedWorkItems.GBKIntoKeyedWorkItems.class.getCanonicalName()));
+    }
+  }
+
+  /**
+   * A translator just to vend the URN. This will need to be moved to runners-core-construction-java
```
Comment (aljoscha) [inline runners/flink/src/main/java/org/apache/beam/runners/flink/FlinkStreamingTransformTranslators.java]: This is probably a copy-paste error from the other translators

**Comment by aljoscha** [comment]
This looks good, minus one comment I had about comments. 😉

**Pull request merged**